Source/santasyncservice/SNTSyncManagerTest.mm

```diff
@@ -0,0 +1,192 @@
+/// Copyright 2025 North Pole Security, Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     https://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import "Source/common/SNTSyncConstants.h"
+#import "Source/santasyncservice/SNTPushClientAPNS.h"
+#import "Source/santasyncservice/SNTPushNotifications.h"
+#import "Source/santasyncservice/SNTSyncState.h"
+
+@interface SNTSyncManagerTest : XCTestCase
+@end
+
+@implementation SNTSyncManagerTest
+
+#pragma mark - Test Push Client Interval Updates
+
+// These tests verify that SNTPushClientAPNS correctly updates its interval
+// when handlePreflightSyncState: is called. This is the core functionality
+// that SNTSyncManager relies on to detect interval changes.
+
+- (void)testPushClientAPNSDefaultInterval {
+  // Verify default initialization
+  SNTPushClientAPNS *pushClient = [[SNTPushClientAPNS alloc] initWithSyncDelegate:nil];
+
+  XCTAssertEqual(pushClient.fullSyncInterval, kDefaultPushNotificationsFullSyncInterval,
+                 @"Push client should initialize with default interval");
+}
+
+- (void)testPushClientAPNSIntervalUpdate {
```

Comment:
It might be more useful to have the test pulled up one layer in the stack. All the tests below essentially check that `handlePreflightSyncState`, but doesn't attempt to ensure timers are accounted for.